### PR TITLE
added %matplotlib --list so backends are easily visible

### PR DESCRIPTION
**Pull request by koenvb**
linked to #8444 adding --list argument to show the available backends

### CODE REVIEW

**Comment by koenvb** [comment]
Seems like 

--l 
--li
--lis
--list

are all giving the same result. Dunno exactly why? And if its appropriate

```
In [4]: %matplotlib --l
Available matplotlib backends: ['osx', 'qt4', 'qt5', 'gtk3', 'notebook', 'wx', 'qt', 'nbagg', 'gtk', 'tk', 'inline']

In [5]: %matplotlib --li
Available matplotlib backends: ['osx', 'qt4', 'qt5', 'gtk3', 'notebook', 'wx', 'qt', 'nbagg', 'gtk', 'tk', 'inline']

In [6]: %matplotlib --lis
Available matplotlib backends: ['osx', 'qt4', 'qt5', 'gtk3', 'notebook', 'wx', 'qt', 'nbagg', 'gtk', 'tk', 'inline']

In [7]: %matplotlib --list
Available matplotlib backends: ['osx', 'qt4', 'qt5', 'gtk3', 'notebook', 'wx', 'qt', 'nbagg', 'gtk', 'tk', 'inline']
```

File: IPython/core/magics/pylab.py [hunk 1]
```diff
@@ -84,9 +86,15 @@ def matplotlib(self, line=''):
 
             In [3]: %matplotlib qt
         """
+
         args = magic_arguments.parse_argstring(self.matplotlib, line)
-        gui, backend = self.shell.enable_matplotlib(args.gui)
-        self._show_matplotlib_backend(args.gui, backend)
+
+        if args.list is not None:
```
Comment (Carreau) [inline IPython/core/magics/pylab.py]: My guess is as as you do `store_true ` above `args.list` can only be `True̵̵̵|False` so the `... is None` is alway True, and you get into the if.

**Comment by Carreau** [comment]
Do you want to take a stab at fixing the `--list` thing, or should one of us take over ? (There is no hurry)

**Comment by koenvb** [comment]
Will give it another try. can I just update this one then, or should I create a new pull request?

**Comment by koenvb** [comment]
I was looking into it a bit more and I get a bit confused. It feels like when using --l or --li or --lis
the arg.list is always True and I would expect its only True when -l or --list?

So adding these print statements to the code:
(I do not know how to run a debugger in this case? any workflow help is appreciated)
```
print(line)
args = magic_arguments.parse_argstring(self.matplotlib, line)
print(args)
print(args.list)
```

Gives:

```
In [5]: %matplotlib --l
--l
Namespace(gui=None, list=True)
True
Available matplotlib backends: ['osx', 'qt4', 'qt5', 'gtk3', 'notebook', 'wx', 'qt', 'nbagg', 'gtk', 'tk', 'inline']

In [6]: %matplotlib --li
--li
Namespace(gui=None, list=True)
True
Available matplotlib backends: ['osx', 'qt4', 'qt5', 'gtk3', 'notebook', 'wx', 'qt', 'nbagg', 'gtk', 'tk', 'inline']

In [7]: %matplotlib --lis
--lis
Namespace(gui=None, list=True)
True
Available matplotlib backends: ['osx', 'qt4', 'qt5', 'gtk3', 'notebook', 'wx', 'qt', 'nbagg', 'gtk', 'tk', 'inline']

In [8]: %matplotlib --lisg
--lisg
UsageError: unrecognized arguments: --lisg
```

I would assume that the 
```args = magic_arguments.parse_argstring(self.matplotlib, line) function```
should reject if it is not -l or --list?

But I am probably wrong here...

**Comment by Carreau** [comment]
we use argparse, and it seem to be [a feature](https://docs.python.org/3.5/library/argparse.html#argument-abbreviations-prefix-matching), we could [disable it](https://docs.python.org/3.5/library/argparse.html#allow-abbrev) though.

And Yes, usually you can just push on github it updates the PR.
If its an argparse problem, we can (decide to) fix  it later.

**Comment by Carreau** [comment]
@takluyver a look, you were mentioned in the issue ? do you think list for for backend is fine, or should we refine as bullet points ?

File: IPython/core/magics/pylab.py (hunk 1 above)
Comment (takluyver) [inline IPython/core/magics/pylab.py]: I think this should just be `if args.list`, then you shouldn't need the default above (with `store_true` the default should automatically be False).

**Comment by takluyver** [comment]
I think a list is fine. I've made one minor suggestion inline.

**Comment by koenvb** [comment]
I updated the line and added something to the docs. 
Let me know if all is fine!

File: IPython/core/magics/pylab.py [hunk 2]
```diff
@@ -42,6 +42,8 @@ class PylabMagics(Magics):
     @skip_doctest
     @line_magic
     @magic_arguments.magic_arguments()
+    @magic_arguments.argument('-l', '--list', default=False, action='store_true',
```
Comment (takluyver) [inline IPython/core/magics/pylab.py]: You don't actually need the `default=False` here - when you use store_true, the default is automatically False.

Comment (Carreau) [inline IPython/core/magics/pylab.py]: "Explicit is better than implicit" might be objected :-)

Comment (takluyver) [inline IPython/core/magics/pylab.py]: I think `'store_true'` makes it clear enough what's going on. All the examples in the argparse docs using store_true don't bother with a default.

Comment (HerlonNascimento) [inline IPython/core/magics/pylab.py]: 'store_true' makes it clear enough what's going on

**Comment by takluyver** [comment]
Thanks @koenvb

**Pull request merged**